.github/workflows/publish_release.yml

```diff
@@ -0,0 +1,79 @@
+name: Publish a new release
+
+on:
+  workflow_dispatch:
+    inputs:
+      isMajor:
+        type: boolean
+        required: false
+        description: This is a new major version
+      message:
+        type: string
+        required: false
+        description: Commit message
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Set up JDK 11
+      uses: actions/setup-java@v3
+      with:
+        java-version: '11'
+        distribution: adopt
+        cache: maven
+
+    - name: Build with Maven
+      run: mvn --batch-mode -DskipTests package --update-snapshots verify
```

Comment:
Make a composition action to use those jobs in release and in build. Check other jobs if there is other possible compositions